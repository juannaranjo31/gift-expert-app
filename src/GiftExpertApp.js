import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GiftGrid } from './components/GiftGrid';

const GiftExpertApp = ( {defaultCategories = [] } ) => {

    const [Categories, setCategories] = useState(defaultCategories);

    return <>
        <h2>GiftExpertApp</h2>
        <hr />

        <AddCategory setCategories={setCategories} />

        <ol>
            {
                Categories.map( category => {
                    return <GiftGrid key={category} category= {category}/>
                })
            }
        </ol>
    </>
}

export default GiftExpertApp;