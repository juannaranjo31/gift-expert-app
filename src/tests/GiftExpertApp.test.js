import React from 'react'
import { shallow } from "enzyme";
import GiftExpertApp from '../GiftExpertApp';

describe('Pruebas sobre < GiftExpertApp />', () => {

    test('Debe reenderizarse correctamente', () => {

        const wrapper = shallow(<GiftExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorias', () => {

        const categories = ['Dragon Ball', 'Naruto'];
        const wrapper = shallow(<GiftExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GiftGrid').length).toBe(categories.length);

    });

});
