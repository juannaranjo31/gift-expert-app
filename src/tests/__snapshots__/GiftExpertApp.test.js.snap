// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre < GiftExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GiftGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GiftGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas sobre < GiftExpertApp /> Debe reenderizarse correctamente 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
